### ISSUE TITLE
Vapor 4 - Cannot attach siblings relation to unsaved model

### ISSUE THREAD

**Issue by davuthdv** [vapor/vapor#2091]
I got an error `"Cannot attach siblings relation to unsaved model."` even though both the models to and being attached to are already saved. 

### Steps to reproduce

I have 2 tables: `Product` & `Tag`, and a pivot `ProductTag`. The model is pretty much the same as the `Planet` & `Tag` in the Fluent's release note.

If i do this:
```swift
return product.create(on: database)
    .flatMap { tag.create(on: database) }
    .flatMap { product.$tags.attach(tag, on: database) }
```
it's fine. However, if change product to array of product like this:
```swift
return [product].create(on: database)
    .flatMap { tag.create(on: database) }
    .flatMap { product.$tags.attach(tag, on: database) }
```
I got an error: `"Cannot attach siblings relation to unsaved model."` Does that mean calling `create` on array of model won't wait until all items complete?

#### Additional info:

And this also works:
```swift
return [product].create(on: database)
    .flatMap { tag.create(on: database) }
    .flatMap { _ -> EventLoopFuture<Product> in
        Product.query(on: database).filter(\.$name == "Product Name").first().unwrap(or: Abort(.internalServerError))
    }
    .flatMap { product in
        product.$tags.attach(tag, on: database)
    }
```

### Environment

* Vapor Framework version: 4.0.0-beta.1
* Fluent version: 4.0.0-beta.1
* fluent-postgres-driver: 2.0.0-beta.1.1
* OS version: macOS Catalina

**Comment by jhoughjr** [comment]
I'm just a dumb ass bug but I'm dealing with this issue as well. Very bad code smell to have to refetch to finish populating imo. Might not scale well.

**Comment by tanner0101** [comment]
@davuthdv @jhoughjr are you using `UUID` ids? If so, it should be possible to do this without re-fetching. I will take a look at fixing that. This will not be possible for database-generated IDs though since there's no way to get those identifiers back during a batch creation. 

For db-generated ids, you could do:
```swift
products.map { $0.create(...) }.flatten(...).flatMap {
    // each product will have its id
}
```

**Comment by jhoughjr** [comment]
Currently my app is cobbled together from examples, and I barely know what I'm doing. I may not need siblings

**Comment by mcdappdev** [comment]
@tanner0101 totally possible that I'm reading this wrong but does that mean that it's not possible to bulk save items and have the ID be passed back appropriately? (Using db-generated)

**Comment by tanner0101** [comment]
@mcdappdev yeah that's correct. MySQL and SQLite have methods for getting the last autoincremented ID which is what normal `create` uses. AFAIK there's nothing like that for batch inserts.

**Comment by mcdappdev** [comment]
Ahh interesting. Ok so for times when you need the saved ID of each model you should still do the .map { $0.save } and then flatten? Otherwise use the batch save?

**Comment by tanner0101** [comment]
Yeah that or batch insert + fetch. The latter would probably be faster since it's only 2 queries as opposed to `n`.

**Comment by mcdappdev** [comment]
Got it. What would that second fetch look like?

**Comment by tanner0101** [comment]
Oh, good point. That would only work if you have some way of uniquely identifying only the newly inserted models. Probably `map { $0.create(...) }` is best, then if you can't use UUID ids.

**Comment by mcdappdev** [comment]
Ok cool! Thanks!
